src/Symfony/Component/HttpFoundation/HeaderUtils.php

```diff
@@ -211,12 +211,19 @@ private static function groupParts(array $matches, string $separators): array
         $parts = [];
         if ($partSeparators) {
             foreach ($partMatches as $matches) {
-                $parts[] = self::groupParts($matches, $partSeparators);
+                $parts[] = self::groupParts($matches, $partSeparators, false);
             }
         } else {
             foreach ($partMatches as $matches) {
                 $parts[] = self::unquote($matches[0][0]);
             }
+
+            if (!$first && 2 < count($parts)) {
+                $parts = [
```

Comment:
I believe this implode should be moved before the `unquote`.
ie. `foo="a=b"`. 
+ the loop at line 217 use only `matxh[0][0]` don't we loose something here?

Reply:
IMO this works as expected, but I added a test anyway